### DISCUSSION TITLE
Share artifact between two Workflows/CronWorkflows

### DISCUSSION THREAD

**Discussion by mj3c** [argoproj/argo-workflows#5350] [Q&A] [answered]
Hey Argo community!

What I am trying to do is share an artifact between two separate workflows.

One example would be to have a Workflow (let's call it A) that I run today, and another workflow B that I run tomorrow but I want it to fetch an artifact from A.

Another example would be to have a CronWorkflow A that executes daily, and then a regular Workflow B that fetches an artifact from the latest execution of A.

I assume this is not possible? From what I found, sharing artifacts is only possible between steps within a singular Workflow/CronWorkflow. I know something like that can be done, in general, but I was wondering if it's possible to do this with native Argo artifacts.

**Comment by simster7** [comment, marked as answer]
This is certainly possible. You need to set up an Artifact repository. You can use a cloud provider (e.g. S3) or set up a repository in your own cluster (e.g. the team uses Minio). You can take a look at the possible Artifact repository types [here](https://github.com/argoproj/argo-workflows/blob/master/docs/configure-artifact-repository.md).

Once you have an artifact repository configured, your workflows can simply read and write the artifacts from said repository, instead of passing it around different steps. Take a look at these examples on how to get and put artifacts from an artifact repository:

* https://github.com/argoproj/argo-workflows/blob/master/examples/input-artifact-s3.yaml
* https://github.com/argoproj/argo-workflows/blob/master/examples/output-artifact-s3.yaml

**Reply by mj3c** [reply]
Thank you so much for a quick answer!

I have been going around the documentation quite a lot, but somehow I managed to overlook this... I was glued to the whole artifact-passing example too much, as well as the "global" outputs. What you suggested is still a tad bit different than I imagined but this can cover what I need.

For the second (CronWorkflow) scenario I described, one solution would be to define two same artifacts, one with a generated key and another with a fixed "latest" key or something, like so:
```yaml
    outputs:
      artifacts:
      - name: message
        path: /tmp
      - name: message-latest
        path: /tmp
        s3:
          key: path/to/latest/stuff.txt.tgz
```
This is in order to have a history of past artifacts as well... if you define it with a fixed path only, it will always overwrite the previous artifact.

Thanks again!
